### ISSUE TITLE
[RabbitMQ] Health Check fails for SSL/TLS connections

### ISSUE THREAD

**Issue by rwkarg** [Xabaril/AspNetCore.Diagnostics.HealthChecks#324]
This commit https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/commit/3c149151d3ccb044995e1df9526e702da7ba744d removes passing in the `_sslOptions`, even if they are specified. This prevents a health check from being successful to an SSL/TLS endpoint on a broker.

Behavior is as expected in version 3.0.0, prior to this commit.

**Comment by unaizorrilla** [comment]
Hi @rwkarg 

You can specify ssloption on ConnectionFactory definition. This works for you?
```

        public async Task be_healthy_if_rabbitmq_is_available_using_iconnectionfactory()
        {
            var connectionString = @"amqp://localhost:5672";

            var factory = new ConnectionFactory()
            {
                Uri = new Uri(connectionString),
                AutomaticRecoveryEnabled = true,
                Ssl= new SslOption(serverName: "localhost", enabled: false)
            };

            var webHostBuilder = new WebHostBuilder()
            .UseStartup<DefaultStartup>()
            .ConfigureServices(services =>
            {
                services
                    .AddHealthChecks()
                    .AddRabbitMQ(sp=>factory,  tags: new string[] { "rabbitmq" });
            })
            .Configure(app =>
            {
                app.UseHealthChecks("/health", new HealthCheckOptions()
                {
                    Predicate = r => r.Tags.Contains("rabbitmq")
                });
            });

            var server = new TestServer(webHostBuilder);

            var response = await server.CreateRequest($"/health")
                .GetAsync();

            response.StatusCode
                .Should().Be(HttpStatusCode.OK);
        }
```

**Comment by rwkarg** [comment]
This constructor:
https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/3c149151d3ccb044995e1df9526e702da7ba744d/src/HealthChecks.RabbitMQ/RabbitMQHealthCheck.cs#L17 
allows passing of SslOptions which get saved internally, but the previously mentioned PR removes any usage of that option.
Behavior in 3.0.0 would use the specified SslOption which is not surprising. The 3.0.1 package will accept the SslOption in the constructor but then ignore it which is surprising.

**Comment by rwkarg** [comment]
Is there any more information about what "fix a minor issue with ssl options" is from the previous checkin? (It doesn't look like there was a PR for that change to link). It appears that the change was just to ignore the passed in `sslOption` which seems to just be wrong, unless there is some context I'm missing.

Otherwise, the fix seems to be reverting the code back to actually using the `sslOption` if it has been passed in, as it was doing previously.

**Comment by unaizorrilla** [comment]
Hi @rwkarg 

If you need to use ssl options you can use the overload with ConnectionFactory, this does not work for you?

```
            var connectionString = @"amqp://localhost:5672";

            var factory = new ConnectionFactory()
            {
                Uri = new Uri(connectionString),
                AutomaticRecoveryEnabled = true,
                Ssl= new SslOption(serverName: "localhost", enabled: false)
            };

            var webHostBuilder = new WebHostBuilder()
            .UseStartup<DefaultStartup>()
            .ConfigureServices(services =>
            {
                services
                    .AddHealthChecks()
                    .AddRabbitMQ(sp=>factory,  tags: new string[] { "rabbitmq" });
            })
```

**Comment by rwkarg** [comment]
No. Updating all of our services across the company is not a desirable option.

What is the resistance to fixing the issue with the `.AddRabbitMQ(string, SslOption)` method? It works as expected in 3.0.0.

**Comment by unaizorrilla** [comment]
Is not "resistance", i need some time to do that!! PR is welcome of course

**Comment by rwkarg** [comment]
I'm happy to submit a PR. I just need to know what "fix a minor issue with ssl options" refers to. If I just revert the change that happened in 3.0.1 then I will reintroduce the minor issue.

**Comment by unaizorrilla** [comment]
let me check the changes

**Comment by unaizorrilla** [comment]
Hi @rwkarg 

I don't see more than removed ssloption. PR is welcome, and try to check if sslOptions is honored when method is executed and not used default

![imagen](https://user-images.githubusercontent.com/1630008/69657725-2110ea80-107b-11ea-9d09-5027bdcb7c62.png)

**Comment by rwkarg** [comment]
I'm taking a look at the change and while verifying it, it seems like the DI extension method has been updated to use a factory which will recreate a new RabbitMQHealthCheck each time a check is made which defeats the purpose of having RabbitMQHealthCheck reuse it's internal connection.

I think I'll initially just add back the SslOption support as a minimal change and then tackle actually reusing a single connection in a separate PR.

**Comment by sungam3r** [comment]
Fixed in #347.
